### PR TITLE
Reformatting

### PR DESCRIPTION
**Pull request by kelshadarby**
**In iteration 3 branch**
Create methods for iteration 3:

- Create hashes for each method
- Iterate over hashes to find applicable info with the team id as the key
- Find average or percent for each value array in the hash
- Find max or min for average or percent
- Use key from max or min to find team name

Create modules:

- Create hash_builder method
- Create min and max by for hashes in module
- Create math methods (percent and average)

**In this branch**
Convert league stats to a module:

- Convert league stats class to a module
- Add tests to stat tracker test
- Change all object in test to stat tracker objects
- Merge Margo's best and worst fans methods and helper methods into file

### CODE REVIEW

File: lib/modules/hashable.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  end
+
+  def hash_key_max_by(hash)
+    hash.key(hash.values.max)
+  end
+
+  def hash_key_min_by(hash)
+    hash.key(hash.values.min)
+  end
+end
```
Comment (margoflewelling) [inline lib/modules/hashable.rb:15]: good idea making these modules, could definitely refactor some of the methods to use these

File: lib/modules/league_statistics.rb [hunk 2]
```diff
@@ -0,0 +1,171 @@
+require './lib/game'
+require './lib/modules/calculable'
+require './lib/modules/hashable'
+
+module LeagueStatistics
+  include Calculable
+  include Hashable
```
Comment (margoflewelling) [inline lib/modules/league_statistics.rb:7]: can include modules in a module!

File: lib/modules/league_statistics.rb [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+      wins = (results.find_all { |result| result == "WIN"})
+      percent(wins.length.to_f, results.length)
+    end
+  end
+
+  def winningest_team
+    find_team_names(hash_key_max_by(percent_wins))
+  end
+
+  # =====================================================================
```
Comment (margoflewelling) [inline lib/modules/league_statistics.rb:123]: probably delete this at some point

File: lib/modules/league_statistics.rb [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  end
+
+  def average_visiting_teams_and_goals
+    visiting_teams_and_goals.transform_values do |goals|
+      average(goals.sum.to_f, goals.size)
+    end
+  end
+
+  def highest_scoring_visitor
+    find_team_names(hash_key_max_by(average_visiting_teams_and_goals))
```
Comment (chrisstankus99) [inline lib/modules/league_statistics.rb:79]: Awesome job making a hash in our module. I can already see where else we could use it.

**Comment by foleymichelle9** [comment]
Alt Text

**Comment by foleymichelle9** [comment]
Alt Text

**Pull request merged**